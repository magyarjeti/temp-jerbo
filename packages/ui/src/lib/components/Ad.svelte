<script lang="ts">
	import { resize } from 'svelte-resize-observer-action';

	interface Props {
		id: string;
		adIndicator?: boolean | undefined;
	}

	let { id, adIndicator = undefined }: Props = $props();

	let formattedId = $derived(`ap-${id}`);
	let showIndicator = $state(adIndicator ?? false);

	function handleResize(entry: ResizeObserverEntry) {
		showIndicator = adIndicator !== false && entry.contentRect.height > 0;
	}
</script>

<div class="contents">
	<div class="contents" id={formattedId} class:ad={showIndicator} use:resize={handleResize}></div>
</div>

<style>
	.ad {
		text-align: center;
		z-index: 0;
	}

	.ad::after {
		content: 'hirdet√©s';
		display: 'block';
		font-size: '12px';
		text-transform: uppercase;
		font-family: sans-serif;
		text-align: 'center';
		font-family: 'boxed';
		margin-top: '4px';
		text-shadow: '1px 1px 0px #fff';
	}
</style>
