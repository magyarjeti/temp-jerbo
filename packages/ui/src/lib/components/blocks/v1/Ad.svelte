<script lang="ts">
	import type { AdBlock } from '@packages/api/sdk/v1/schema';
	import Ad from '../../Ad.svelte';

	interface Props {
		data?: AdBlock;
	}

	let { data }: Props = $props();
</script>

<Ad id={data?.params?.id || ''} adIndicator={!!data?.params?.adIndicator || false}></Ad>
