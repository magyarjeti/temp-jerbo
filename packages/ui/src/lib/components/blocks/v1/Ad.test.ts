import { render } from '@testing-library/svelte';
import { expect, test } from 'vitest';
import Ad from './Ad.svelte';
import '../../../tests/mocks';

test('Ad snapshot', () => {
	const { container } = render(Ad);
	expect(container).toMatchSnapshot();
});
