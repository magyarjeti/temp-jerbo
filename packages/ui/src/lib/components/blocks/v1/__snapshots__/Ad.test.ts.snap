// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Ad snapshot 1`] = `
<div>
  <div
    class="contents"
  >
    <div
      class="contents svelte-223tdn"
      id="ap-"
    />
  </div>
</div>
`;
